[Bug]: Panic by type of atomicErr

### 🐞 Describe the Bug
The following error is displayed where atomicErr is used.

```bash
panic: sync/atomic: store of inconsistently typed value into Value

goroutine 344 [running]:
sync/atomic.(*Value).Store(0xc012d1b5b0, {0xbd4a20, 0xc012eaa6c0})
        /opt/hostedtoolcache/go/1.23.4/x64/src/sync/atomic/value.go:78 +0xd5
github.com/cresplanex/bloader/internal/runner.run.func2({{0xc00908e1d0, 0x8}, {0xc00908e3b0, 0x10}, {0xc02537c150, 0x17}, 0xc000260500, 0xc000260520, 0x0, {0x0, ...}, ...})
        /home/runner/work/bloader/bloader/internal/runner/flow.go:749 +0xa85
created by github.com/cresplanex/bloader/internal/runner.run in goroutine 1
        /home/runner/work/bloader/bloader/internal/runner/flow.go:693 +0x1775
```

### 🔍 Steps to Reproduce
Steps to reproduce the behavior:
1. Run flow runner

### 🤔 Expected Behavior
No Panic

### 📋 Additional Context

### 🌍 Environment
- OS: 
  ```bash
  Linux *** ****-**** #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
  ```
- Go version: 1.23
- CLI version: v1.0.1

--- a/internal/runner/base.go
+++ b/internal/runner/base.go
@@ -331,7 +331,7 @@ func (e BaseExecutor) Execute(
 			}
 			return fmt.Errorf("failed to connect to slave: %w", err)
 		}
-		atomicErr := atomic.Value{}
+		var atomicErr atomic.Pointer[syncError]
 		var wg sync.WaitGroup
 		for _, slave := range validSlaveConnect.Slaves {
 			wg.Add(1)
@@ -350,7 +350,7 @@ func (e BaseExecutor) Execute(
 					e.TargetFactor,
 					e.Store,
 				); err != nil {
-					atomicErr.Store(err)
+					atomicErr.Store(&syncError{Err: err})
 					e.Logger.Error(ctx, "failed to handle response: %v",
 						logger.Value("error", err))
 					cancel()
@@ -359,15 +359,10 @@ func (e BaseExecutor) Execute(
 			}(slHandler)
 		}
 		wg.Wait()
-		if err := atomicErr.Load(); err != nil {
-			if err, ok := err.(error); ok {
-				e.Logger.Error(ctx, "failed to find error",
-					logger.Value("error", err), logger.Value("on", "Flow"))
-				return err
-			}
-			e.Logger.Error(ctx, "failed to find unknown error",
-				logger.Value("on", "Flow"))
-			return fmt.Errorf("failed to find unknown error")
+		if syncErr := atomicErr.Load(); syncErr != nil {
+			e.Logger.Error(ctx, "failed to find error",
+				logger.Value("error", syncErr.Err), logger.Value("on", "Execute"))
+			return syncErr.Err
 		}
 		e.Logger.Info(ctx, "connected to slave node")
 	case RunnerKindFlow:

--- a/internal/runner/error.go
+++ b/internal/runner/error.go
@@ -1,0 +1,5 @@
+package runner
+
+type syncError struct {
+	Err error
+}

--- a/internal/runner/flow.go
+++ b/internal/runner/flow.go
@@ -678,7 +678,7 @@ func run(
 			}
 		}
 	} else {
-		atomicErr := atomic.Value{}
+		var atomicErr atomic.Pointer[syncError]
 		var wg sync.WaitGroup
 		sem := make(chan struct{}, concurrency)
 		for i, executor := range executors {
@@ -697,7 +697,7 @@ func run(
 					if err := preExecutor.castFunc(ctx); err != nil {
 						log.Error(ctx, fmt.Sprintf("failed to cast[%d]", i),
 							logger.Value("error", err), logger.Value("on", "Flow"))
-						atomicErr.Store(err)
+						atomicErr.Store(&syncError{Err: err})
 						cancel()
 					}
 				}()
@@ -729,7 +729,8 @@ func run(
 						NewDefaultEventCasterWithBroadcaster(preExecutor.eventCaster),
 					)
 					if err != nil {
-						atomicErr.Store(err)
+						fmt.Printf("type file failed to execute flow[%d], %v(type: %T)\n", i, err, err)
+						atomicErr.Store(&syncError{Err: err})
 						log.Error(ctx, fmt.Sprintf("failed to execute flow[%d]", i),
 							logger.Value("error", err), logger.Value("on", "Flow"))
 						cancel()
@@ -746,7 +747,8 @@ func run(
 						flows[i],
 					)
 					if err != nil {
-						atomicErr.Store(err)
+						fmt.Printf("type slCmd failed to execute flow[%d], %v(type: %T)\n", i, err, err)
+						atomicErr.Store(&syncError{Err: err})
 						log.Error(ctx, fmt.Sprintf("failed to execute flow[%d]", i),
 							logger.Value("error", err), logger.Value("on", "Flow"))
 						cancel()
@@ -773,7 +775,8 @@ func run(
 						broadCastMap,
 					)
 					if err != nil {
-						atomicErr.Store(err)
+						fmt.Printf("type flow failed to execute flow[%d], %v(type: %T)\n", i, err, err)
+						atomicErr.Store(&syncError{Err: err})
 						log.Error(ctx, fmt.Sprintf("failed to execute flow[%d]", i),
 							logger.Value("error", err), logger.Value("on", "Flow"))
 						cancel()
@@ -791,15 +794,10 @@ func run(
 
 		close(sem)
 
-		if err := atomicErr.Load(); err != nil {
-			if e, ok := err.(error); ok {
-				log.Error(ctx, "failed to find error",
-					logger.Value("error", e), logger.Value("on", "Flow"))
-				return e
-			}
-			log.Error(ctx, "failed to find unknown error",
-				logger.Value("on", "Flow"))
-			return fmt.Errorf("failed to find unknown error")
+		if syncErr := atomicErr.Load(); syncErr != nil {
+			log.Error(ctx, "failed to find error",
+				logger.Value("error", syncErr.Err), logger.Value("on", "Flow"))
+			return syncErr.Err
 		}
 
 		return nil
@@ -961,7 +959,7 @@ func slaveCmdRun(
 		}
 	}
 
-	atomicErr := atomic.Value{}
+	var atomicErr atomic.Pointer[syncError]
 	var wg sync.WaitGroup
 	for i, executor := range slaveExecutors {
 		wg.Add(1)
@@ -971,7 +969,7 @@ func slaveCmdRun(
 
 			err := preExecutor.exec(ctx, log)
 			if err != nil {
-				atomicErr.Store(err)
+				atomicErr.Store(&syncError{Err: err})
 				log.Error(ctx, fmt.Sprintf("failed to execute flow[%d]", i),
 					logger.Value("error", err), logger.Value("on", "Flow"))
 				return
@@ -981,15 +979,10 @@ func slaveCmdRun(
 
 	wg.Wait()
 
-	if err := atomicErr.Load(); err != nil {
-		if e, ok := err.(error); ok {
-			log.Error(ctx, "failed to find error",
-				logger.Value("error", e), logger.Value("on", "Flow"))
-			return e
-		}
-		log.Error(ctx, "failed to find unknown error",
-			logger.Value("on", "Flow"))
-		return fmt.Errorf("failed to find unknown error")
+	if syncErr := atomicErr.Load(); syncErr != nil {
+		log.Error(ctx, "failed to find error",
+			logger.Value("error", syncErr.Err), logger.Value("on", "slaveCmdRun"))
+		return syncErr.Err
 	}
 
 	return nil

--- a/internal/runner/massexec.go
+++ b/internal/runner/massexec.go
@@ -495,7 +495,7 @@ func (r ValidMassExec) runHTTP(
 	}
 
 	var wg sync.WaitGroup
-	atomicErr := atomic.Value{}
+	var atomicErr atomic.Pointer[syncError]
 	startChan := make(chan struct{})
 	for _, executor := range threadExecutors {
 		wg.Add(1)
@@ -509,7 +509,7 @@ func (r ValidMassExec) runHTTP(
 			defer wg.Done()
 			defer close(exec.ReqTermChan)
 			if err := exec.Execute(ctx, log, startChan); err != nil {
-				atomicErr.Store(err)
+				atomicErr.Store(&syncError{Err: err})
 				log.Error(ctx, "failed to execute",
 					logger.Value("error", err), logger.Value("id", exec.ID))
 				return
@@ -519,15 +519,10 @@ func (r ValidMassExec) runHTTP(
 	close(startChan)
 	wg.Wait()
 
-	if err := atomicErr.Load(); err != nil {
-		if e, ok := err.(error); ok {
-			log.Error(ctx, "failed to find error",
-				logger.Value("error", e), logger.Value("on", "MassExecuteBatch"))
-			return e
-		}
-		log.Error(ctx, "failed to find unknown error")
-
-		return fmt.Errorf("failed to find error")
+	if syncErr := atomicErr.Load(); syncErr != nil {
+		log.Error(ctx, "failed to find error",
+			logger.Value("error", syncErr.Err), logger.Value("on", "ValidMassExec.runHTTP"))
+		return syncErr.Err
 	}
 
 	return nil

